staging/src/k8s.io/apiserver/pkg/authentication/authenticatorfactory/requestheader.go

```diff
@@ -24,8 +28,9 @@ type RequestHeaderConfig struct {
 	// ExtraHeaderPrefixes are the head prefixes to check (case-insentively) for filling in
 	// the user.Info.Extra.  All values of all matching headers will be added.
 	ExtraHeaderPrefixes []string
-	// ClientCA points to CA bundle file which is used verify the identity of the front proxy
-	ClientCA string
+	// VerifyOptionFn are the options for verifying incoming connections using mTLS.  Generally this points to CA bundle file which is used verify the identity of the front proxy.
+	// It may produce different options at will.
+	VerifyOptionFn x509request.VerifyOptionFunc
```

Comment:
Can we add a bool or an enum and make this in ApplyTo? I'd like to keep these options structs serializable and deep equatable if possible.

Reply:
> Can we add a bool or an enum and make this in ApplyTo? I'd like to keep these options structs serializable and deep equatable if possible.

The end goal is to wire up to a configmap which has no logical ClientCA and would use a client which cannot be serialized.

Reply:
To add some more clarity, this instance is an intermediate form that is embedded into non-serializeable structs.  This is not the struct that we serialize for config or directly wire options against.

Reply:
Yes, your write. I thought this was Options for some reason.